### PR TITLE
Add Transit Gateway Support and Vault Integration for Dynamic Value Association in PowerVS e2e CI Job

### PR DESCRIPTION
**Pull request by aman4433**
This PR introduces the following updates to the PowerVS e2e CI Jobs:

**Transit Gateway (TG) Support:**
Added support for Transit Gateway in the e2e CI Jobs to enable network configuration in environments using TG.
Replaced the cloud connection flag with the following new parameters:

`--e2e-powervs-power-edge-router=true` 
`--e2e-powervs-transit-gateway=<dynamically-fetched-value>`
`--e2e-powervs-transit-gateway-location=<dynamically-fetched-value>`

**Vault Integration for Dynamic Value Association:**
Integrated Vault to dynamically associate values.
This enhancement provides secure storage and management of sensitive values that are dynamically fetched and used in the CI Jobs.

### CODE REVIEW

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

File: ci-operator/step-registry/hypershift/powervs/run-e2e/hypershift-powervs-run-e2e-chain.yaml [hunk 1]
```diff
@@ -4,6 +4,41 @@ chain:
     - as: run-e2e
       cli: latest
       commands: |-
+        # Load values from existing-resources.json if not provided by the environment
+        if [[ -z "${POWERVS_GUID}" ]]; then
+          POWERVS_GUID=$(jq -r '.cloudInstanceID' "${CLUSTER_PROFILE_DIR}/existing-resources.json")
+          export POWERVS_GUID
```
Comment (dharaneeshvrd) [inline ci-operator/step-registry/hypershift/powervs/run-e2e/hypershift-powervs-run-e2e-chain.yaml:10]: After defining them in env section, do you need to export here?

**Review by dharaneeshvrd — COMMENTED** [review]
Let's test with 4.17 as well!

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.17-periodics-e2e-powervs-ovn

File: ci-operator/step-registry/hypershift/powervs/run-e2e/hypershift-powervs-run-e2e-chain.yaml (hunk 1 above)
Comment (aman4433) [reply ci-operator/step-registry/hypershift/powervs/run-e2e/hypershift-powervs-run-e2e-chain.yaml:10]: I believe if variables are set in the env section, exporting isn’t needed. But if they're dynamically assigned (e.g., from existing-resources.json), export is necessary for subprocesses. Let me know if I'm on track!

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.17-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Comment by aman4433** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-powervs-ovn

**Pull request merged**